### PR TITLE
merge functions from ios into ios_config

### PR DESCRIPTION
**Pull request by privateip**
- merge changes from ios shared module functions into ios_config.
- add src argument to provide path to config file
- add new choice to match used to ignore current running config
- add update argument with choices merge or check
- add backup argument to backup current running config to control host
- add defaults argument to control collection of config with or withoutdefaults
- add save argument to save current running config to startup config
- add state argument to control state of config file
- deprecated force argument, use match=none instead

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @privateip. This module is going into community review.

We notice that you are one of the maintainers (@privateip, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
This PR is a work-in-progress. @privateip, when you're ready to have this PR reviewed, please remove WIP from the title of the PR.

File: network/ios/ios_config.py [hunk 1]
```diff
@@ -122,17 +98,44 @@
...
-      - This arugment will either cause or prevent the changed commands
-        from being sent to the remote device.  The set to true, the
-        remote IOS device will be configured with the updated commands
-        and when set to false, the remote device will not be updated.
+      - The force argument instructs the module to not consider the
+        current devices running-config.  When set to true, this will
+        cause the module to push the contents of I(src) into the device
+        without first checking if already configured.
+      - Note this argument should be considered deprecated.  To achieve
+        the equivalient, set the match argument to none.  This argument
```
Comment (gundalow) [inline network/ios/ios_config.py]: set the C(match=none).

I wonder if the deprecated notice should be on the first line so people will spot it

**Comment by gundalow** [comment]
shipit

**Pull request merged**